java/client/src/main/java/glide/api/BaseClient.java

```diff
@@ -807,6 +812,43 @@ public CompletableFuture<Long> xtrim(@NonNull String key, @NonNull StreamTrimOpt
         return commandManager.submitNewCommand(XTrim, arguments, this::handleLongResponse);
     }
 
+    @Override
+    public CompletableFuture<Map<String, Map<String, String[]>>> xread(
+            @NonNull Map<String, String> keysAndIds) {
+        return xread(keysAndIds, StreamReadOptions.builder().build());
+    }
+
+    @Override
+    public CompletableFuture<Map<String, Map<String, String[]>>> xread(
+            @NonNull Map<String, String> keysAndIds, StreamReadOptions options) {
+        String[] arguments = options.toArgs(keysAndIds);
+        return commandManager.submitNewCommand(
+                XRead,
+                arguments,
+                response ->
+                        /**
+                         * Receive a MAP of OBJECTS, the object is a 2-D array of objects that needs to be
```

Comment:
LOLWUT
looks like a conversion we do in glide core - why not to do it there?
It is actually a 4D array which could (should?) be converted to
```
    Map<String, Map<String, Map<String, String>>>
          ^            ^           ^        ^
        stream         id         key     value
```

even if we don't do conversion in glide core (why don't?) it should be moved to a separate function from there

Reply:
if we do that conversion in glide core, `handleMapResponse` probably should do all rest

Reply:
documentation says about different data types in resp2/3 so you have to go to glide